ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/IndexMappingTool.java

```diff
@@ -86,75 +88,80 @@ public Object parse(Object o) {
 
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
-        @SuppressWarnings("unchecked")
-        List<String> indexList = parameters.containsKey("index")
-            ? gson.fromJson(parameters.get("index"), List.class)
-            : Collections.emptyList();
-        if (indexList.isEmpty()) {
-            @SuppressWarnings("unchecked")
-            T empty = (T) ("There were no results searching the index parameter [" + parameters.get("index") + "].");
-            listener.onResponse(empty);
-            return;
-        }
+        try {
+            List<String> indexList = new ArrayList<>();
+            if (StringUtils.isNotBlank(parameters.get("index"))) {
```

Comment:
will the parameters.get("index") return null instead of blank string? should we handle null cases too?

Reply:
if that's the case, the next check for the contains key is not needed